csharp/ql/src/semmle/code/csharp/Attribute.qll

```diff
@@ -51,9 +54,37 @@ class Attribute extends TopLevelExprParent, @attribute {
   /** Gets the element that this attribute is attached to. */
   Attributable getTarget() { attributes(this, _, result) }
 
-  /** Gets the `i`th argument of this attribute. */
+  /**
+   * Gets the `i`th argument of this attribute. This includes both constructor
+   * arguments and named arguments.
+   */
   Expr getArgument(int i) { result = this.getChildExpr(i) }
 
+  /**
+   * Gets the `i`th constructor argument of this attribute. For example, only
+   * `true` is a constructor argument in
+   *
+   * ```
+   * MyAttribute[true, Foo = 0]
+   * ```
+   */
+  Expr getConstructorArgument(int i) {
```

Comment:
We probably don't need this (or, make `private`) if we make `getArgument()` more powerful and implement `getArgumentForParameter`.

Reply:
See my comment above.